src/ucp/proto/rndv.c

```diff
@@ -9,10 +9,10 @@
 #endif
 
 #include "rndv.h"
-#include "tag_rndv.h"
-#include "tag_match.inl"
-#include "offload.h"
 
+#include <ucp/tag/tag_rndv.h>
+#include <ucp/tag/tag_match.inl>
+#include <ucp/tag/offload.h>
```

Comment:
so proto rndv includes tag rndv? looks like invalid layering

Reply:
yes, because proto rndv contains common RTS handler for TAG and AM rndvs. Currently only tag handler is invoked, AM will be added with new API

Reply:
can we move the relevant headers to proto?
tag should include proto headers , not other way around..

Reply:
we have to call tag/AM specific handler based on the RTS->flags.

Reply:
try to include only ucp/tag/tag_rndv.h